### PR TITLE
add media query listener

### PR DESCRIPTION
**Pull request by mangoischke**
update mediaQuery and it’s test to matchMediaQuery

| Q             | A
| ------------- | --------------------------------------------------------------------- |
| Bug fix?      |no                                                                |
| New feature?  | yes                   |
| Improvement?  |no  |
| BC breaks?    |no                                                                |
| Deprecations? |no    |
| Docs PR       | **missing**                                   |

### CODE REVIEW

File: tests/cases/dom/events/matchMediaQuery.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+            };
+        };
+
+        matchMediaQuery("screen and (min-width: 400px)", (matches) => {
+            assert.ok(matches, "The listener was called with a matching query.");
+        });
+
+        window.matchMedia = realMatchMedia;
+    }
+);
```
Comment (apfelbox) [inline tests/cases/dom/events/matchMediaQuery.js]: Can you maybe add a test:

with `immediate = false`

Comment (mangoischke) [reply tests/cases/dom/events/matchMediaQuery.js]: the first test is an `immediate = false` test

**Comment by apfelbox** [comment]
I might have an idea for a slightly different implementation / extension, so please have some patience.

File: tests/cases/dom/events/matchMediaQuery.js (hunk 1 above)
Comment (apfelbox) [reply tests/cases/dom/events/matchMediaQuery.js]: Ah, perfekt!

**Comment by apfelbox** [comment]
So I finally was able to add my integration. I pushed it in this PR, to see the changes.

The idea is the following: get away from every component having its own media query, but instead having a single:

`lib/media-queries.ts`:
```ts
export const onContentWidth = mediaQueryMatcher("screen and (min-width: 1128px");
export const onStructureWidth = mediaQueryMatcher("screen and (min-width: 1440px");
// ...
```

And every component can then hook into these global media queries. You would have the definitions in a global file anyway, so we can just put the complete matcher there.

Usage for example in a hook:

```jsx
let [isContentWidth, setContentWidth] = useState(onContentWidth.matches());

useEffect(() => {
    onContentWidth.on(setContentWidth);
    return () => onContentWidth.off(setContentWidth);
}, []);
```

**Comment by apfelbox** [comment]
I also added support for `onOff()`, so the example above can be written as:

```jsx
let [isContentWidth, setContentWidth] = useState(onContentWidth.matches());

useEffect(() => onContentWidth.onOff(setContentWidth), []);
```

**Pull request merged**